.github/workflows/stacks-blockchain-api.yml

```diff
@@ -10,6 +10,8 @@ on:
       - '**/CHANGELOG.md'
       - '**/package.json'
   pull_request:
+    types:
+      - opened
```

Comment:
Instead of only having CI run when a PR is opened, ideally it would run on every new commit to the PR as well to ensure the merged code is being used in a build. If it's only running on branch commits and not PRs, then the images will potentially be missing new commits added downstream in the `develop` or `master` branch.

[See how this is configured here](https://github.com/hirosystems/gh-actions-example/blob/main/.github/workflows/ci.yml#L41-L43), where `pull_request` isn't filtered, but the _branches_ are filtered to only run on long-lived branches, like main/master and develop. If you're looking to avoid duplicate builds on a single commit, this would be a good setup for that.

Reply:
My original idea was to leave this 'PR opened' event plus the 'all branches' setting above which builds on every push for every branch (regardless of PR status). Do you think this could work as well?

I'd like to avoid building only on `develop`/`master` because sometimes it's very useful to push a branch to Github (without a PR) and do some very quick tests against the full test suite.

Reply:
I can see how that's useful, but I think the important part missing is that if a PR is opened, there always needs to be a build that reflects the most recent _merge_ commit. Building on a PR vs a feature branch is quite different in CI; they're technically two different commit SHAs. If CI is only configured to run when the PR is first opened, and not on subsequent merge commits, then it opens us up to potential bugs that would only be caught post-merge.

If you want it to build on every commit to a branch, that should be fine to keep that in there as well. The only downside is that commits will trigger two workflows and consume more GH runners (this is a shared resource throughout the org). Configuring it like this also means Github will consume resources on commits to branches that aren't ready for testing yet, and opening a draft PR I find is a good way to signal to Github that it's at a state where it can be tested and you'd find the results useful in some way.

Reply:
Gotcha, thanks for the clarification 👍